godot-core/src/meta/property_info.rs

```diff
@@ -179,6 +180,48 @@ impl PropertyInfo {
         }
     }
 
+    /// Consumes self, moving the values into `sys::GDExtensionPropertyInfo` pointer.
+    ///
+    /// # Safety
+    ///
+    /// * `property_info_ptr` must be valid.
+    pub(crate) unsafe fn move_into_property_info_ptr(
+        self,
+        property_info_ptr: *mut sys::GDExtensionPropertyInfo,
+    ) {
+        let class_name = if self.class_name == ClassName::none() {
+            (*property_info_ptr).class_name
+        } else {
+            sys::SysPtr::force_mut(self.class_name.string_sys())
+        };
+        property_info_ptr.write(self.property_sys());
+        (*property_info_ptr).class_name.write(*class_name);
+
+        // Decreasing the refcount on refcounted types (such as StringName or GString)
+        // is a responsibility of the `property_info_ptr`'s owner.
+        std::mem::forget(self);
```

Comment:
With the current implementation, Godot cannot know when to decrease the refcounts though -- you simply overwrite the pointers. So I think this would leak memory.